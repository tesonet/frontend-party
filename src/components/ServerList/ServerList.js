import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import { ROUTE_LOGIN } from '../../constants/routes';
import copy from './ServerList.json';

class ServerList extends Component {
  constructor(props) {
    super(props);

    this.checkAuthToken = this.checkAuthToken.bind(this);
  }

  componentWillMount() {
    this.checkAuthToken(this.props.token);
  }

  componentWillReceiveProps(nextProps) {
    this.checkAuthToken(nextProps.token);
  }

  checkAuthToken(token) {
    if (!token) {
      this.props.push(ROUTE_LOGIN);
    }
  }

  render() {
    return (
      <div>
        <header>
          <h1>{copy.welcome}</h1>
        </header>
        <p>
          {copy.body}
        </p>
      </div>
    );
  }
}

ServerList.propTypes = {
  push: PropTypes.func.isRequired,
  token: PropTypes.string,
};

const mapStateToProps = store => {
  const { auth: { token } } = store;

  return { token };
};

const mapDispatchToProps = dispatch => ({
  push: path => dispatch(push(path)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ServerList);
