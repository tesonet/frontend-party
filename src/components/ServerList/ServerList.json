{
  "welcome": "Welcome to the Frontend Party App",
  "body": "Let's have a fun of a paragraph here, shall we?"
}
