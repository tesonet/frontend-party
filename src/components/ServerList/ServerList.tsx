import React from 'react';
import Branding from '../Branding';
import Button from '../Button';
import Table from '../Table';
import { RouteComponentProps } from 'react-router';

const ServerList: React.FC<RouteComponentProps> = props => {
  const handleLogout = () => {
    localStorage.removeItem('token');
    props.history.replace('/');
  };

  const token = localStorage.getItem('token');
  if (!token) props.history.replace('/');

  return (
    <div>
      <div className={'flex justify-between items-center my-5 mx-4'}>
        <Branding textColor={'text-black'} />
        <Button
          text={'Logout'}
          onClick={handleLogout}
          classes={'text-black border-solid border-2 border-blue-700'}
        />
      </div>
      <Table />
    </div>
  );
};

export default ServerList;
