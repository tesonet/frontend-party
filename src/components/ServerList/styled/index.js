import Header from './Header';
import SignOutButton from './SignOutButton';
import StyledContainer from './StyledContainer';
import StyledListContainer from './StyledListContainer';
import StyledListHeading from './StyledListHeading';
import StyledList from './StyledList';
import StyledRow from './StyledRow';
import StyledLogo from './StyledLogo';

export {
  Header,
  SignOutButton,
  StyledContainer,
  StyledListContainer,
  StyledListHeading,
  StyledList,
  StyledRow,
  StyledLogo,
};
