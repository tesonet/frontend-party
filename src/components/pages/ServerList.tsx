import React from 'react';
import Container from '../styled/Container';
import List, {
  ListRow,
  ListItem,
  ListHeaderRow,
} from '../common/List';
import Header from '../common/Header';
import ScrollArea from '../common/ScrollArea';

export default function ServerList() {
  const serverList = [
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
    {
      name: 'aaa',
      distance: 66,
      id: 1,
    },
    {
      name: 'bbb',
      distance: 66,
      id: 2,
    },
  ];
  return (
    <Container>
      <Header />
      <List>
        <ListHeaderRow>
          <ListItem>SERVER</ListItem>
          <ListItem>DISTANCE</ListItem>
        </ListHeaderRow>
        <ScrollArea
          style={{
            width: '100%',
            height: 'calc(100vh - ( 113px + 25px + 20px + 16px ))',
          }}
        >
          {serverList.map(server => (
            <ListRow key={server.id}>
              <ListItem>{server.name}</ListItem>
              <ListItem>{server.distance}</ListItem>
            </ListRow>
          ))}
        </ScrollArea>
      </List>
    </Container>
  );
}
