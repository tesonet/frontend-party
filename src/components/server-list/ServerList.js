import React, { useEffect, Fragment } from "react";
import PropTypes from "prop-types";

import ServerListHeader from "../server-list-header";

import styles from "./styles.css";

const ServerList = ({ getServerListAction, serverList }) => {
  useEffect(() => {
    getServerListAction();
  }, [getServerListAction]);
  return (
    <Fragment>
      <ServerListHeader />
      <div>
        <div className={`${styles.tableRow} ${styles.tableHeader}`}>
          <h6 className={`${styles.text} ${styles.tableHeaderText}`}>server</h6>
          <h6 className={`${styles.text} ${styles.tableHeaderText}`}>
            distance
          </h6>
        </div>
        <div className={styles.tableBody}>
          {serverList.map(server => (
            <div
              key={server.name + server.distance}
              className={styles.tableRow}
            >
              <p className={styles.text}>{server.name}</p>
              <p className={styles.text}>{server.distance}</p>
            </div>
          ))}
        </div>
      </div>
    </Fragment>
  );
};

ServerList.propTypes = {
  getServerListAction: PropTypes.func.isRequired,
  serverList: PropTypes.arrayOf(
    PropTypes.shape({
      distance: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired
    })
  ).isRequired
};

export default ServerList;
