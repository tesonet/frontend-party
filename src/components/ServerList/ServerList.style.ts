import styled, { css } from 'styled-components'
import { headerHeight } from 'components/Header/Header.style'

const listTableHeight = '3.25rem'

const fixedHeaderStyle = css`
  position: fixed;
  top: ${headerHeight};
  right: 0;
  left: 0;
  height: ${listTableHeight};
`

const activeHeaderItemStyle = css`
  color: ${({ theme }) => theme.colors.atlantis};
  font-weight: 400;
`

export const ServerListWrapper = styled.div``

export const ServerItemWrapper = styled.div<{ header?: boolean }>`
  width: 100%;
  background-color: ${({ theme, header }) =>
    header ? theme.colors.backgroundGray : theme.colors.white};
  display: flex;
  justify-content: space-between;
  font-size: 0.875rem;
  padding: 1rem;
  color: ${({ theme, header }) => (header ? theme.colors.dustyGray : theme.colors.textGray)};
  border-top: 1px solid ${({ theme, header }) => (header ? theme.colors.borderColor : 0)};
  border-bottom: 1px solid ${({ theme }) => theme.colors.borderColor};
  text-transform: ${({ header }) => (header ? 'uppercase' : 'none')};
  font-weight: 300;
  position: static;

  ${({ header }) => header && fixedHeaderStyle};
`
export const Items = styled.div`
  position: relative;
  top: ${listTableHeight};
  height: calc(100vh - ${headerHeight} - 3.625rem);
  overflow-y: auto;
`
export const HeaderItem = styled.div<{ active?: boolean }>`
  cursor: pointer;
  display: flex;

  ${({ active }) => active && activeHeaderItemStyle};
`
