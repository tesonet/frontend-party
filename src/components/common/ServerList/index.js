import React, {useEffect} from 'react'
import PropTypes from 'prop-types'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import styled from 'styled-components'
import {
  List,
  ListHeader,
  ListHeaderItem,
  ListHeaderText,
  ListItem,
  ListItemText,
  Icon,
  Spinner,
  Button,
  Alert,
  Box,
} from '../../../ui'
import * as selectors from '../../../modules/servers/selectors'
import * as serverActions from '../../../modules/servers/actions'

const SortIcon = styled(Icon)`
  fill: currentColor;
  opacity: ${props => (props.show ? 1 : 0)};
  transition: transform 0.3s, opacity 0.3s;
  transform: rotate(
    ${props => {
      return props.direction === 'down' ? 180 : 0
    }}deg
  );
`

const SortableHeaderItem = styled(ListHeaderItem)`
  cursor: pointer;
  &:hover ${SortIcon} {
    opacity: 1;
  }
`

const getSortDirection = (key, currentKey, order) => {
  if (key === currentKey) {
    return order
  }

  return 'asc'
}

const ServerList = ({loading, error, items, sortParams, actions}) => {
  const {key: sortKey, order} = sortParams

  const handleRequestSort = property => {
    let sortDirection = getSortDirection(property, sortKey, order)

    if (property === sortKey) {
      sortDirection = sortDirection === 'asc' ? 'desc' : 'asc'
    }

    actions.updateServersSortParams({
      key: property,
      order: sortDirection,
    })
  }

  useEffect(() => {
    actions.fetchServers()
  }, [actions])

  const renderContent = () => {
    return (
      <List>
        <ListHeader>
          <SortableHeaderItem
            onClick={() => {
              handleRequestSort('name')
            }}
            data-test-id="server-list-header-name">
            <ListHeaderText>
              Server
              <SortIcon
                name="arrow-up"
                ml={1}
                direction={
                  getSortDirection('name', sortKey, order) === 'desc'
                    ? 'down'
                    : 'up'
                }
                show={sortKey === 'name'}
              />
            </ListHeaderText>
          </SortableHeaderItem>

          <SortableHeaderItem
            onClick={() => {
              handleRequestSort('distance')
            }}
            data-test-id="server-list-header-distance">
            <ListHeaderText>
              <SortIcon
                name="arrow-up"
                mr={1}
                direction={
                  getSortDirection('distance', sortKey, order) === 'desc'
                    ? 'down'
                    : 'up'
                }
                show={sortKey === 'distance'}
              />
              Distance
            </ListHeaderText>
          </SortableHeaderItem>
        </ListHeader>
        {items.map(item => {
          return (
            <ListItem
              key={`${item.name}-${item.distance}`}
              data-test-id="servers-list-item">
              <ListItemText>{item.name}</ListItemText>
              <ListItemText>{item.distance}</ListItemText>
            </ListItem>
          )
        })}
      </List>
    )
  }

  const renderError = () => {
    return (
      <Box maxWidth={300} p={3} mx="auto">
        <Alert variant="danger" mb={2}>
          {error?.response?.message || error?.message}
        </Alert>
        <Button fullWidth onClick={actions.fetchServers}>
          Reload list
        </Button>
      </Box>
    )
  }

  if (loading) {
    return <Spinner size="large" mt={3} mx="auto" />
  }

  if (error) {
    return renderError()
  }

  return renderContent()
}

ServerList.propTypes = {
  loading: PropTypes.bool,
  // eslint-disable-next-line react/forbid-prop-types
  error: PropTypes.object,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      distance: PropTypes.number,
    }),
  ),
  sortParams: PropTypes.shape({
    key: PropTypes.string,
    order: PropTypes.oneOf(['asc', 'desc']),
  }).isRequired,
  actions: PropTypes.shape({
    fetchServers: PropTypes.func,
    updateServersSortParams: PropTypes.func,
  }).isRequired,
}

ServerList.defaultProps = {
  loading: false,
  error: null,
  items: [],
}

const mapStateToProps = state => ({
  loading: selectors.getIsLoading(state),
  error: selectors.getError(state),
  items: selectors.getSortedSeverList(state),
  sortParams: selectors.getServerSortParams(state),
})

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(serverActions, dispatch),
})

export default connect(mapStateToProps, mapDispatchToProps)(ServerList)
