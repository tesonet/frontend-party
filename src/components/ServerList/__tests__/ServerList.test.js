import React from 'react';
import { shallow } from 'enzyme';

import ServerList from '../ServerList';

const withFormik = jest.fn(); // eslint-disable-line no-unused-vars

describe('ServerList component', () => {
  const token = 'auth-token-12345';
  const list = [];
  const props = {
    store: {
      getState: () => ({
        auth: {
          token,
          isLoading: false,
          error: null,
        },
        serverList: { list },
      }),
      subscribe: jest.fn(),
      dispatch: jest.fn(),
    },
    retrieveServerList: jest.fn(),
    push: jest.fn(),
    logout: jest.fn(),
    token,
    list,
  };

  it('should render the ServerList component.', () => {
    const renderedComponent = shallow(<ServerList {...props} />).dive();

    expect(renderedComponent).toMatchSnapshot();
  });
});
