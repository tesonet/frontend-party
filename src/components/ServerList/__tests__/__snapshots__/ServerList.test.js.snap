// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerList component should render the ServerList component. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ServerList
    list={Array []}
    logout={[Function]}
    push={[Function]}
    retrieveServerList={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": undefined,
          }
    }
    token="auth-token-12345"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Container
          fluid={true}
          tag="div"
>
          <HelmetWrapper
                    defer={true}
                    encodeSpecialCharacters={true}
          >
                    <title>
                              Server List
                    </title>
          </HelmetWrapper>
          <styled.header>
                    <StyledLogo />
                    <SignOutButton
                              onClick={[Function]}
                    />
          </styled.header>
</Container>,
        <styled.div>
          <span>
                    Server
          </span>
          <span>
                    Distance
          </span>
</styled.div>,
        <styled.div>
          <styled.div>
                    
          </styled.div>
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <HelmetWrapper
              defer={true}
              encodeSpecialCharacters={true}
>
              <title>
                            Server List
              </title>
</HelmetWrapper>,
            <styled.header>
              <StyledLogo />
              <SignOutButton
                            onClick={[Function]}
              />
</styled.header>,
          ],
          "fluid": true,
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <title>
                Server List
</title>,
              "defer": true,
              "encodeSpecialCharacters": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Server List",
              },
              "ref": null,
              "rendered": "Server List",
              "type": "title",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <StyledLogo />,
                <SignOutButton
                  onClick={[Function]}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <span>
              Server
</span>,
            <span>
              Distance
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Server",
            },
            "ref": null,
            "rendered": "Server",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Distance",
            },
            "ref": null,
            "rendered": "Distance",
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div>
            
</styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Container
            fluid={true}
            tag="div"
>
            <HelmetWrapper
                        defer={true}
                        encodeSpecialCharacters={true}
            >
                        <title>
                                    Server List
                        </title>
            </HelmetWrapper>
            <styled.header>
                        <StyledLogo />
                        <SignOutButton
                                    onClick={[Function]}
                        />
            </styled.header>
</Container>,
          <styled.div>
            <span>
                        Server
            </span>
            <span>
                        Distance
            </span>
</styled.div>,
          <styled.div>
            <styled.div>
                        
            </styled.div>
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <HelmetWrapper
                defer={true}
                encodeSpecialCharacters={true}
>
                <title>
                                Server List
                </title>
</HelmetWrapper>,
              <styled.header>
                <StyledLogo />
                <SignOutButton
                                onClick={[Function]}
                />
</styled.header>,
            ],
            "fluid": true,
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <title>
                  Server List
</title>,
                "defer": true,
                "encodeSpecialCharacters": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Server List",
                },
                "ref": null,
                "rendered": "Server List",
                "type": "title",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <StyledLogo />,
                  <SignOutButton
                    onClick={[Function]}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span>
                Server
</span>,
              <span>
                Distance
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Server",
              },
              "ref": null,
              "rendered": "Server",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Distance",
              },
              "ref": null,
              "rendered": "Distance",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.div>
              
</styled.div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
